From 45d3c4fc724081726142c515b69582d991475ca6 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 19 Mar 2020 10:04:46 +0000
Subject: [PATCH] ARM: dts: bcm271x: Use a53 pmu, drop RPI364

The upstream bcm2837.dtsi uses cortex-a53-pmu, so we can do the same
but with a fallback to the cortex-a7-pmu which is supported by the
32-bit kernel.

Now that we're using the natural fallback mechanism of compatible
strings, the RPI364 macro no longer serves any purpose - remove it.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2710.dtsi                        | 6 +-----
 arch/arm64/boot/dts/broadcom/bcm2710-rpi-2-b.dts      | 2 --
 arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b-plus.dts | 2 --
 arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b.dts      | 2 --
 arch/arm64/boot/dts/broadcom/bcm2710-rpi-cm3.dts      | 2 --
 arch/arm64/boot/dts/broadcom/bcm2711-rpi-4-b.dts      | 2 --
 6 files changed, 1 insertion(+), 15 deletions(-)

diff --git a/arch/arm/boot/dts/bcm2710.dtsi b/arch/arm/boot/dts/bcm2710.dtsi
index 4e47480dd933..e7e5c913f1d1 100644
--- a/arch/arm/boot/dts/bcm2710.dtsi
+++ b/arch/arm/boot/dts/bcm2710.dtsi
@@ -5,11 +5,7 @@
 	compatible = "brcm,bcm2837", "brcm,bcm2836";
 
 	arm-pmu {
-#ifdef RPI364
-		compatible = "arm,armv8-pmuv3", "arm,cortex-a7-pmu";
-#else
-		compatible = "arm,cortex-a7-pmu";
-#endif
+		compatible = "arm,cortex-a53-pmu", "arm,cortex-a7-pmu";
 	};
 
 	soc {
diff --git a/arch/arm64/boot/dts/broadcom/bcm2710-rpi-2-b.dts b/arch/arm64/boot/dts/broadcom/bcm2710-rpi-2-b.dts
index 116cdbf94b9b..36ecea71f0ef 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2710-rpi-2-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2710-rpi-2-b.dts
@@ -1,3 +1 @@
-#define RPI364
-
 #include "../../../../arm/boot/dts/bcm2710-rpi-2-b.dts"
diff --git a/arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b-plus.dts b/arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b-plus.dts
index d9242ff77079..22fc6a82f2a9 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b-plus.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b-plus.dts
@@ -1,3 +1 @@
-#define RPI364
-
 #include "../../../../arm/boot/dts/bcm2710-rpi-3-b-plus.dts"
diff --git a/arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b.dts b/arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b.dts
index deb33441da95..4cacc5b72ae3 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2710-rpi-3-b.dts
@@ -1,3 +1 @@
-#define RPI364
-
 #include "../../../../arm/boot/dts/bcm2710-rpi-3-b.dts"
diff --git a/arch/arm64/boot/dts/broadcom/bcm2710-rpi-cm3.dts b/arch/arm64/boot/dts/broadcom/bcm2710-rpi-cm3.dts
index 1c2560017c02..e1e13784cff6 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2710-rpi-cm3.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2710-rpi-cm3.dts
@@ -1,3 +1 @@
-#define RPI364
-
 #include "../../../../arm/boot/dts/bcm2710-rpi-cm3.dts"
diff --git a/arch/arm64/boot/dts/broadcom/bcm2711-rpi-4-b.dts b/arch/arm64/boot/dts/broadcom/bcm2711-rpi-4-b.dts
index 1fd86f81f542..bf69a4b0b172 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2711-rpi-4-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2711-rpi-4-b.dts
@@ -1,3 +1 @@
-#define RPI364
-
 #include "../../../../arm/boot/dts/bcm2711-rpi-4-b.dts"
-- 
2.17.1

